pallets/gas/src/lib.rs

```diff
@@ -0,0 +1,498 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use self::imbalances::{NegativeImbalance, PositiveImbalance};
+use common::{DAGBasedLedger, Origin};
+use frame_support::{dispatch::DispatchError, pallet_prelude::*, traits::Imbalance};
+pub use pallet::*;
+pub use primitive_types::H256;
+use scale_info::TypeInfo;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[derive(Clone, Decode, Debug, Encode, MaxEncodedLen, TypeInfo)]
+pub enum ValueOrigin {
+    External(H256),
+    Local(H256),
+}
+
+#[allow(clippy::derivable_impls)]
+// this cannot be derived, despite clippy is saying this!!
+impl Default for ValueOrigin {
+    fn default() -> Self {
+        ValueOrigin::External(H256::default())
+    }
+}
+
+#[derive(Clone, Default, Decode, Debug, Encode, MaxEncodedLen, TypeInfo)]
+pub struct ValueNode {
```

Comment:
Does this intentionally duplicate `common::ValueNode`?

Reply:
Yes. All the related stuff in `common` is about to be removed (it's already not being used anywhere in the code and has been kept for testing purpose to be able to compare two implementations results).